### PR TITLE
 O3-1557: run e2e tests on the 3.x branch at openmrs-distro-reference application

### PR DESCRIPTION
**Pull request by jnsereko**
I strongly stand by us running all tests rather than smoke tests.  I have honestly failed to sieve out what test is more worth than the other.  

I will also create another workflow to run smoke tests so devs can optionally skip (ignore) the all e2e tests workflow in case they are overwhelmed by the reds while merging. 

cc @dkayiwa @gracepotma @ibacher

### CODE REVIEW

File: .github/workflows/refapp-3x-vitals-and-biometrics.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        run: docker compose up -d --build
+      - name: Wait for Openmrs instance to start
+        run: while [[ "$(curl -s -o /dev/null -w ''%{http_code}'' http://localhost:8080/openmrs/login.htm)" != "200" ]]; do sleep 10; done
+      - name: Clone tests
+        uses: actions/checkout@v3
+        with:
+          repository: openmrs/openmrs-test-3refapp
+          path: openmrs-test-3refapp
+      - name: Installing Repo Dependencies
+        run: cd openmrs-test-3refapp && yarn install
```
Comment (ibacher) [inline .github/workflows/refapp-3x-vitals-and-biometrics.yml:28]:
```suggestion
        run: cd openmrs-test-3refapp && yarn install --immutable
```

File: .github/workflows/refapp-3x-vitals-and-biometrics.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+      - name: Run e2e tests
+        run: cd openmrs-test-3refapp/packages/cypress && yarn refappCypress
+      - name: Upload screen recordings of failed tests
+        if: ${{ failure() }}
+        uses: actions/upload-artifact@v2
+        with:
+          name: Screen recordings of failed tests
+          path: packages/cypress/cypress/videos
```
Comment (ibacher) [inline .github/workflows/refapp-3x-vitals-and-biometrics.yml:36]:
```suggestion
          path: openmrs-test-3refapp/packages/cypress/cypress/videos
```

**Review by ibacher — CHANGES_REQUESTED** [review]
I think this PR is mis-targetted. In particular, this should run as part of the Bamboo job rather than GitHub Actions here. Triggering this as part of GitHub Actions means it will only run on PRs and merges _to this repo_. Most such merges are adding metadata to the RefApp and these tests add very little value there. However, the Bamboo job is currently triggered on every commit to the RefApp frontend repo, so it will be able to catch breaking changes after they are committed.

PS Thanks @jnsereko for all your work on this. It's really just a matter of making sure we run these tests at the right point.

**Comment by jnsereko** [comment]
thank you so much for the response @ibacher 
I kindly request you and @dkayiwa to help me an we follow up on this at https://openmrs.slack.com/archives/CP343JH1R/p1665404314967979

**Comment by jayasanka-sack** [comment]
Closing this PR as it has already been addressed in https://issues.openmrs.org/browse/O3-2493

**Pull request closed without merging**